ios/Classes/MapboxMapController.swift

```diff
@@ -652,6 +652,49 @@ class MapboxMapController: NSObject, FlutterPlatformView, MGLMapViewDelegate, Ma
             guard let source = self.mapView.style?.source(withIdentifier: imageSourceId) else { return }
             self.mapView.style?.removeSource(source)
             result(nil)
+        case "style#getLayers":
+            let layers = style.styleManager.getStyleLayers()
+            let layerIds = layers.map { $0.id }
+            result(layerIds)
+        case "style#showLayer":
+            guard let arguments = methodCall.arguments as? [String: Any] else { return }
+            guard let imageLayerId = arguments["imageLayerId"] as? String else { return }
+
+            //Check for duplicateLayer error
+            if (self.mapView.style?.layer(withIdentifier: imageLayerId)) == nil {
+                result(FlutterError(code: "missingLayer", message: "Layer does not exists", details: "Can't show missing layer with imageLayerId: \(imageLayerId)" ))
+                return
+            }
+
+            guard let layer = self.mapView.style?.layer(withIdentifier: imageLayerId) else { return }
```

Comment:
the check above probably  is not required as you can just handle it in the else here